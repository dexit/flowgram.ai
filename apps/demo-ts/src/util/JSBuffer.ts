export class JSBuffer {
  buffer: string[] = []

  constructor(public indent = '') {}

  clear(): JSBuffer {
    this.buffer = []
    return this
  }

  push(...args: string[]): JSBuffer {
    this.buffer.push(...args)
    return this
  }

  pushLine(...args: string[]): JSBuffer {
    this.buffer.push(...args, '\n')
    return this
  }

  toJs(separator: string = ''): string {
    // TODO js string perf
    // FIXME last line indent
    return (
      this.buffer
        .join(separator)
        .split('\n')
        .map((line) => `${this.indent}${line}`)
        // .map((line) => (line !== '\n' ? `${this.indent}${line}` : line))
        .join('\n')
    )
  }

  toString(): string {
    return this.toJs()
  }
}
