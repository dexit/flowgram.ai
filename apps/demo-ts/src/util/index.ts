export * from './log'
export * from './JSBuffer'
