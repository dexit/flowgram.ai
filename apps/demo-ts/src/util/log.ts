import debug from 'debug'

export const info = debug('log:info')
export const log = info
export const warn = debug('log:warn')
export const error = debug('log:error')
