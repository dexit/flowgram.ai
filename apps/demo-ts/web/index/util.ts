
export const codeMirrorBaseConfig: CodeMirror.EditorConfiguration = Object.freeze({
  mode: { name: 'javascript', json: true },
  lineNumbers: true,
  lineWrapping: false,
  theme: 'twilight',
  foldGutter: true,
  gutters: ['CodeMirror-linenumbers', 'CodeMirror-foldgutter'],
  readOnly: true,
  cursorBlinkRate: -1,
})