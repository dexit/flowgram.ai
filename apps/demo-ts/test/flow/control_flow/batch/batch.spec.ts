import { runCases } from '../../util'

runCases(
  [
    'batch',
    { name: 'batch', args: { nums: [] } },
    { name: 'batch', args: { nums: [-3, -2, -1, 1, 2, 3] } },
    'batch_outputs',
  ],
  __dirname
)
