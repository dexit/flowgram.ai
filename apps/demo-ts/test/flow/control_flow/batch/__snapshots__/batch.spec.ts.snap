// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flow batch | {} 1`] = `
"export async function __main(__args) {
  //=== start@start | {level:1,blockIndex:0}
  const start = {}
  start.outputs = {
    nums: [1, 2, 3],
    sum: 0,
  }
  Object.assign(start.outputs, __args)
  //=== batch@batch | {level:1,blockIndex:1}
  const batch = {}
  batch.outputs = {}
  batch.localVariables = {
    item: 0,
    index: 0,
  }
  // concurrency: 3
  batch.__array = start.outputs.nums
  batch.__array.forEach((__item, __index) => {
    batch.localVariables.item = __item
    batch.localVariables.index = __index
    //=== assign_sum@variable | {level:2,blockIndex:0,parent:'batch'}
    const assign_sum = {}
    assign_sum.outputs = {}
    start.outputs.sum = start.outputs.sum + batch.localVariables.item
  }) // batch@batch
  //=== end@end | {level:1,blockIndex:2}
  const end = {}
  end.data = {
    sum: start.outputs.sum,
  }
  return end.data
}
"
`;

exports[`flow batch | {} 2`] = `
{
  "error": {
    "code": "",
    "msg": "",
  },
  "sum": 6,
}
`;

exports[`flow batch | {nums:[]} 1`] = `
"export async function __main(__args) {
  //=== start@start | {level:1,blockIndex:0}
  const start = {}
  start.outputs = {
    nums: [1, 2, 3],
    sum: 0,
  }
  Object.assign(start.outputs, __args)
  //=== batch@batch | {level:1,blockIndex:1}
  const batch = {}
  batch.outputs = {}
  batch.localVariables = {
    item: 0,
    index: 0,
  }
  // concurrency: 3
  batch.__array = start.outputs.nums
  batch.__array.forEach((__item, __index) => {
    batch.localVariables.item = __item
    batch.localVariables.index = __index
    //=== assign_sum@variable | {level:2,blockIndex:0,parent:'batch'}
    const assign_sum = {}
    assign_sum.outputs = {}
    start.outputs.sum = start.outputs.sum + batch.localVariables.item
  }) // batch@batch
  //=== end@end | {level:1,blockIndex:2}
  const end = {}
  end.data = {
    sum: start.outputs.sum,
  }
  return end.data
}
"
`;

exports[`flow batch | {nums:[]} 2`] = `
{
  "error": {
    "code": "",
    "msg": "",
  },
  "sum": 0,
}
`;

exports[`flow batch | {nums:[-3,-2,-1,1,2,3]} 1`] = `
"export async function __main(__args) {
  //=== start@start | {level:1,blockIndex:0}
  const start = {}
  start.outputs = {
    nums: [1, 2, 3],
    sum: 0,
  }
  Object.assign(start.outputs, __args)
  //=== batch@batch | {level:1,blockIndex:1}
  const batch = {}
  batch.outputs = {}
  batch.localVariables = {
    item: 0,
    index: 0,
  }
  // concurrency: 3
  batch.__array = start.outputs.nums
  batch.__array.forEach((__item, __index) => {
    batch.localVariables.item = __item
    batch.localVariables.index = __index
    //=== assign_sum@variable | {level:2,blockIndex:0,parent:'batch'}
    const assign_sum = {}
    assign_sum.outputs = {}
    start.outputs.sum = start.outputs.sum + batch.localVariables.item
  }) // batch@batch
  //=== end@end | {level:1,blockIndex:2}
  const end = {}
  end.data = {
    sum: start.outputs.sum,
  }
  return end.data
}
"
`;

exports[`flow batch | {nums:[-3,-2,-1,1,2,3]} 2`] = `
{
  "error": {
    "code": "",
    "msg": "",
  },
  "sum": 0,
}
`;

exports[`flow batch_outputs | {} 1`] = `
"export async function __main(__args) {
  //=== start@start | {level:1,blockIndex:0}
  const start = {}
  start.outputs = {}
  Object.assign(start.outputs, __args)
  //=== batch@batch | {level:1,blockIndex:1}
  const batch = {}
  batch.outputs = {
    sum: 0,
  }
  batch.localVariables = {
    item: 0,
    index: 0,
  }
  // concurrency: 1
  batch.__array = [1, 2, 3, 4, 5]
  batch.__array.forEach((__item, __index) => {
    batch.localVariables.item = __item
    batch.localVariables.index = __index
    //=== variable@variable | {level:2,blockIndex:0,parent:'batch'}
    const variable = {}
    variable.outputs = {}
    batch.outputs.sum = batch.outputs.sum + batch.localVariables.item
  }) // batch@batch
  //=== end@end | {level:1,blockIndex:2}
  const end = {}
  end.data = {
    res: batch.outputs.sum,
  }
  return end.data
}
"
`;

exports[`flow batch_outputs | {} 2`] = `
{
  "error": {
    "code": "",
    "msg": "",
  },
  "res": 15,
}
`;
